### ISSUE TITLE
missing face in Magnum::Primitives::cylinderSolid with texture coords

### ISSUE THREAD

**Issue by affa0** [mosra/magnum#386]
if CylinderFlag::GenerateTextureCoords and CylinderFlag::CapEnds is used with cylinderSolid a face is missing.

Can be reproduced with this adapted python example:
removing the primitives.CylinderFlags.GENERATE_TEXTURE_COORDS flag fixes it 

```python
from magnum import *
from magnum import gl, meshtools, primitives, shaders
from magnum.platform.sdl2 import Application

class PrimitivesExample(Application):
    def __init__(self):
        configuration = self.Configuration()
        configuration.title = "Magnum Python Primitives Example"
        Application.__init__(self, configuration)

        gl.Renderer.enable(gl.Renderer.Feature.DEPTH_TEST)
        gl.Renderer.enable(gl.Renderer.Feature.FACE_CULLING)

        #with this one face is missing
        self._mesh = meshtools.compile(primitives.cylinder_solid(1,16,0.5, primitives.CylinderFlags.CAP_ENDS | primitives.CylinderFlags.GENERATE_TEXTURE_COORDS))
        #without GENERATE_TEXTURE_COORDS it seems to work
        #self._mesh = meshtools.compile(primitives.cylinder_solid(1,16,0.5, primitives.CylinderFlags.CAP_ENDS))

        self._shader = shaders.Phong()

        self._transformation = (
            Matrix4.rotation_x(Deg(30.0))@
            Matrix4.rotation_y(Deg(40.0)))
        self._projection = (
            Matrix4.perspective_projection(
                fov=Deg(35.0), aspect_ratio=1.33333, near=0.01, far=100.0)@
            Matrix4.translation(Vector3.z_axis(-10.0)))
        self._color = Color3.from_hsv(Deg(35.0), 1.0, 1.0)
        self._previous_mouse_position = Vector2i()

    def draw_event(self):
        gl.default_framebuffer.clear(gl.FramebufferClear.COLOR|
                                     gl.FramebufferClear.DEPTH)

        self._shader.light_positions = [(7.0, 5.0, 2.5)]
        self._shader.light_colors = [Color3(1.0)]
        self._shader.diffuse_color = self._color
        self._shader.ambient_color = Color3.from_hsv(self._color.hue(), 1.0, 0.3)
        self._shader.transformation_matrix = self._transformation
        self._shader.normal_matrix = self._transformation.rotation_scaling()
        self._shader.projection_matrix = self._projection

        self._mesh.draw(self._shader)
        self.swap_buffers()

    def mouse_release_event(self, event: Application.MouseEvent):
        self._color = Color3.from_hsv(self._color.hue() + Deg(50.0), 1.0, 1.0)
        self.redraw()

    def mouse_move_event(self, event: Application.MouseMoveEvent):
        if event.buttons & self.MouseMoveEvent.Buttons.LEFT:
            delta = 1.0*(
                Vector2(event.position - self._previous_mouse_position)/
                Vector2(self.window_size))
            self._transformation = (
                Matrix4.rotation_x(Rad(delta.y))@
                self._transformation@
                Matrix4.rotation_y(Rad(delta.x)))
            self.redraw()

        self._previous_mouse_position = event.position

exit(PrimitivesExample().exec())```

**Comment by mosra** [comment]
Oh ... not again :laughing: (see #220, #197, ...). I wonder how is it possible that this keeps happening, sorry about that. Will fix that right away.

Btw., thanks for the quick python repro -- I like this way of reproducing issues! :+1:

**Comment by mosra** [comment]
Fixed in 96ea050a74cbf19505131e932789a46d53c67182, with more extensive test cases added in 8c902bec4defaa8bc681e9d6c23d931cf1ea4371.

Thanks for the report!
